domainservice/src/main/kotlin/com/makarytskyi/rentcar/repository/impl/RedisCarRepository.kt

```diff
@@ -0,0 +1,121 @@
+package com.makarytskyi.rentcar.repository.impl
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.makarytskyi.rentcar.config.RedisProperties
+import com.makarytskyi.rentcar.model.MongoCar
+import com.makarytskyi.rentcar.model.patch.MongoCarPatch
+import com.makarytskyi.rentcar.repository.CarRepository
+import io.lettuce.core.RedisException
+import java.net.SocketException
+import java.time.Duration
+import org.springframework.context.annotation.Primary
+import org.springframework.dao.QueryTimeoutException
+import org.springframework.data.redis.RedisConnectionFailureException
+import org.springframework.data.redis.core.ReactiveRedisTemplate
+import org.springframework.stereotype.Repository
+import reactor.core.publisher.Mono
+import reactor.core.scheduler.Schedulers
+import reactor.kotlin.core.publisher.switchIfEmpty
+import reactor.util.retry.Retry
+
+@Repository
+@Primary
+internal class RedisCarRepository(
+    private val mongoCarRepository: CarRepository,
+    private val reactiveRedisTemplate: ReactiveRedisTemplate<String, ByteArray>,
+    private val mapper: ObjectMapper,
+    private val redisProperties: RedisProperties,
+) : CarRepository by mongoCarRepository {
+
+    override fun findById(id: String): Mono<MongoCar> =
+        reactiveRedisTemplate.opsForValue().get(idRedisKey(id))
+            .map { mapper.readValue(it, MongoCar::class.java) }
+            .onErrorResume { Mono.empty() }
+            .switchIfEmpty {
+                mongoCarRepository.findById(id)
+                    .subscribeOn(Schedulers.boundedElastic())
```

Comment:
why do you need subscribeOn here?

Reply:
I forgot to remove it from here, after I moved `subscribeOn` to `setRedisKey`.